src/api/app/helpers/webui/request_helper.rb

```diff
@@ -194,4 +194,62 @@ def next_prev_path(**opts)
       request_show_path(parameters)
     end
   end
+
+  # TODO: find a way to DRY the code related to state badge (used on notifications)
+
+  def bs_request_state_badge(state)
+    content_tag(
+      :span,
+      icon_state_tag(state).concat(state),
+      class: ['badge', "text-bg-#{decode_state_color(state)}", 'ms-1']
+    )
+  end
+
+  private
+
+  def decode_state_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    when :revoked
+      'dismissed'
+    else
+      'dark'
+    end
+  end
+
+  def decode_state_icon(state)
+    case state
+    when :new
+      'code-branch'
+    when :review, :declined, :revoked
+      'code-pull-request'
+    when :superseded
+      'code-compare'
+    when :accepted
+      'code-merge'
+    else
+      'code-fork'
+    end
+  end
```

Comment:
Something for later: probably both these functions could be translated into an `Hash` map of `{ id: value }` for each case.

Reply:
Yeah, I thought about this. I won't do it anything here. But we can improve this later in a follow-up. Also this helper and [BsRequestStateBadgeComponent](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/components/bs_request_state_badge_component.rb) share the logic so, the solution will come once we tackle that.